src/org/javarosa/core/model/instance/TreeElement.java

```diff
@@ -227,6 +228,28 @@ public TreeElement getChild(String name, int multiplicity) {
         return children.get(name);
     }
 
+    /** Returns the first element across descendants that has the specified name. Useful for testing. **/
+    public TreeElement getFirstDescendantWithName(String name) {
+        return getFirstDescendantWithName(this, name);
+    }
+
+    private TreeElement getFirstDescendantWithName(TreeElement node, String name) {
+        if (node.getMultiplicity() != TreeReference.INDEX_TEMPLATE &&
```

Comment:
This is from other search methods such as `findChildrenWithName`. It's important because we want to only identify actual nodes, not templates. Repeats can be templates. This means they're not part of the instance but that their values are used for all repeats that are created. You can read more about `jr:template` at https://opendatakit.github.io/xforms-spec/#primary-instance. So no, it has nothing to do with whether it has children. We want to know whether the node is a template and if so not consider it as a result. I guess the semantic method would be `!node.isTemplate()`.

Reply:
>some static member of TreeReference ...
Do I detect a hint of derision? Heh heh heh. 
I, also, was bothered by that when I first saw it many moons ago.